File: backend/app/infra/db/models/notification.py
"""站内通知表（PRD 5.3.5 表 5.26）。

虽然通知主流程由其他模块完成（事件审核、预警发布等触发），但为保证数据
模型一致性，本骨架先把表建起来——其它组员只需要写 ``NotificationService``。
"""

from __future__ import annotations

from datetime import datetime

from sqlalchemy import BigInteger, ForeignKey, Index, SmallInteger, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.infra.db.base import Base


class Notification(Base):
    __tablename__ = "notifications"
    __table_args__ = (
        Index("idx_notif_user_read_time", "recipient_id", "is_read", "created_at"),
        Index("idx_notif_type", "type"),
        {"comment": "站内通知表 - PRD 5.3.5 表 5.26"},
    )

    notification_id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    recipient_id: Mapped[int] = mapped_column(
        BigInteger,
        ForeignKey("users.user_id", ondelete="RESTRICT"),
        nullable=False,
    )
    type: Mapped[str] = mapped_column(
        String(32),
        nullable=False,
        comment="如 REPORT_RESOLVED / WARNING_PUSH / QUIZ_ASSIGNED",
    )
    title: Mapped[str] = mapped_column(String(128), nullable=False)
    content: Mapped[str] = mapped_column(String(512), nullable=False)
    related_object_type: Mapped[str | None] = mapped_column(String(32), nullable=True)
    related_object_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    is_read: Mapped[int] = mapped_column(
        SmallInteger, nullable=False, default=0, server_default="0"
    )
    created_at: Mapped[datetime] = mapped_column(
        nullable=False, server_default=func.current_timestamp()
    )
    read_at: Mapped[datetime | None] = mapped_column(nullable=True)

File: frontend/src/types/notification.ts
/** 站内通知类型。 */

export interface Notification {
  /** 通知 ID */
  notification_id: string
  /** 通知类型，如 REPORT_RESOLVED / QUIZ_ASSIGNED / WARNING_PUSH */
  type: string
  /** 通知标题 */
  title: string
  /** 通知正文 */
  content: string
  /** 关联对象类型（如 fraud_case / warning_notice），可为 null */
  related_object_type: string | null
  /** 关联对象 ID，可为 null */
  related_object_id: string | null
  /** 是否已读 */
  is_read: boolean
  /** 通知创建时间（ISO 8601） */
  created_at: string
  /** 已读时间，未读时为 null */
  read_at: string | null
}

export interface NotificationPage {
  items: Notification[]
  total: number
  page: number
  size: number
}

export interface UnreadCountResponse {
  count: number
}
